mon-pix/mirage/routes/post-assessments.js

```diff
@@ -4,34 +4,35 @@ import refAssessment from '../data/assessments/ref-assessment';
 
 export default function(schema, request) {
 
-  const answer = JSON.parse(request.requestBody);
-  const courseId = answer.data.relationships.course.data.id;
-
-  const allAssessments = [
-    refAssessment
-  ];
-
-  // TODO: clean legacy
-  const assessments = _.map(allAssessments, function(oneAssessment) {
-    return { id: oneAssessment.data.relationships.course.data.id, obj: oneAssessment };
-  });
+  const requestedAssessment = JSON.parse(request.requestBody);
 
   const newAssessment = {
     'user-id': 'user_id',
     'user-name': 'Jane Doe',
     'user-email': 'jane@acme.com',
-    'certification-number': 'certification-number',
   };
 
-  const assessment = _.find(assessments, { id: courseId });
-  if (assessment) {
-    return assessment.obj;
+  if(requestedAssessment.data.attributes.type === 'SMART_PLACEMENT') {
```

Comment:
J'aurai bien déplace ce if dans la forêt de if plus bas (ça deviendrait un else if). C'est un peu le bordel ce fichier, et j'ai l'impression que ce sera un bordel organisé si tout les if plus bas, gèrent tout les différents types d'assessment. 

`if (assessment) {`
 `   // PLACEMENT ASSESSMENT`
 `   return assessment;`
  `} else if (_.startsWith(courseId, 'null')) {`
   ` // PREVIEW ASSESSMENT`
    `return refAssessment;`
  `} else if (requestedAssessment.data.attributes.type === 'SMART_PLACEMENT') {`
    `// SMART ASSESSMENT`
    `newAssessment.type = 'SMART_PLACEMENT';`
  `} else if (!_.startsWith(courseId, 'rec')) {`
    `// CERTIFICATION ASSESSMENT`
    `newAssessment.type = 'CERTIFICATION';`
    `newAssessment.courseId = 'certification-number';`
    `newAssessment['certification-number'] = 'certification-number';`
  `}`

  `return schema.assessments.create(newAssessment);`

à discuter.